cnvlib/reports.py

```diff
@@ -86,9 +86,9 @@ def gainloss_by_segment(probes, segments, threshold, skip_low=False):
     """
     for segment, subprobes in probes.by_ranges(segments):
         if abs(segment.log2) >= threshold:
-            for (gene, chrom, start, end, _coverage, nprobes
-                ) in group_by_genes(subprobes, skip_low):
-                yield (gene, chrom, start, end, segment.log2, nprobes)
+            for (gene, chrom, start, end, _coverage, nprobes) in \
+                    group_by_genes(subprobes, skip_low):
+                yield segment._replace(gene=gene, start=start, end=end)
```

Comment:
Hi, sorry - I had forgotten the unittests. Now they work fine.

The problem actually was that I was expecting the extra columns in `gainloss_by_gene` where it makes no sense (Impossible to know what kind of aggregation wanted for extra columns).

I fixed that in the following commit: a46acc0

I also fixed the column order to move `gene` to the front as it was before my commits.